core/src/test/java/nl/nn/adapterframework/jta/StatusRecordingTransactionManagerImplementationTest.java

```diff
@@ -64,16 +63,18 @@ public static Collection data() throws NamingException {
 
 	@BeforeClass
 	public static void init() {
-		assumeThat(URLDataSourceFactory.availableDatasources, hasItems(SECONDARY_PRODUCT));
-		for (TransactionManagerType tmt:TransactionManagerType.values()) {
-			tmt.closeConfigurationContext();
-		}
+		assumeThat(URLDataSourceFactory.getAvailableDataSources(), hasItems(SECONDARY_PRODUCT));
+		TransactionManagerType.closeAllConfigurationContexts();
 	}
 
 	@Override
 	public void setup() throws IOException {
 		assumeFalse(transactionManagerType==TransactionManagerType.DATASOURCE);
 		assumeThat(productKey, not(equalTo(SECONDARY_PRODUCT)));
+
+		// Release any hanging commits that might be from previous tests
```

Comment:
Ik zie zo 123 niet waar commits gereleased worden. Onderaan is de release ook weggehaald?

Reply:
De `stop()` method doet dit nu als de `stop` flag van `true` naar `false` wordt gezet.

Voorheen werd niet altijd een `release()` gedaan voordat `stop` op false werd gezet, omdat `stop(false)` dat nu allemaal regelt gaat dit niet meer fout.

Door de Phaser te manipuleren gaan de commit-methods verder met hun commit, en wacht de `stop()` method tot ze klaar zijn voordat hij beëindigd en de transaction manager eventueel gestopt wordt.